papermill/utils.py

```diff
@@ -0,0 +1,22 @@
+from functools import wraps
+import logging
+
+logger = logging.getLogger('papermill.utils')
+
+
+# retry decorator
+def retry(num):
+    def decorate(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            for i in range(num):
+                try:
+                    return func(*args, **kwargs)
+                except Exception as e:
+                    logger.debug('retrying: {}'.format(e))
+            else:
+                raise Exception  # TODO verify what to raise
```

Comment:
I know this was just copied, but let's improve this some. This should `raise e` and the line `e = Exception("No attempts made") should be at the top of `wrapper`.